cmd/releasego/update-azure-linux.go

```diff
@@ -96,19 +116,88 @@ func updateAzureLinux(p subcmd.ParseFunc) error {
+		return fmt.Errorf("failed to get ref: %v", err)
+	}
+
+	newRef := &github.Reference{
+		Ref:    github.String(updateBranch),
+		Object: &github.GitObject{SHA: ref.Object.SHA},
+	}
+
+	if _, _, err = client.Git.CreateRef(ctx, owner, repo, newRef); err != nil {
+		return fmt.Errorf("Failed to create ref: %v", err)
+	}
+
+	updatedFiles := map[string][]byte{
+		golangSpecFilepath:       []byte(golangSpecFileContent),
+		golangSignaturesFilepath: golangSignaturesFileBytes,
+		cgManifestFilepath:       cgManifestBytes,
+	}
+
+	for filePath, newContent := range updatedFiles {
+		// Get the file to update
+		file, _, _, err := client.Repositories.GetContents(ctx, owner, repo, filePath, &github.RepositoryContentGetOptions{Ref: updateBranch})
+		if err != nil {
+			return fmt.Errorf("Failed to get file %q: %v", filePath, err)
+		}
+
+		// Update the file
+		_, _, err = client.Repositories.UpdateFile(ctx, owner, repo, filePath, &github.RepositoryContentFileOptions{
+			Message: github.String(generatePRTitleFromAssets(assets)),
+			Content: newContent,
+			SHA:     file.SHA,
+			Branch:  github.String(updateBranch),
+		})
+		if err != nil {
+			return fmt.Errorf("Failed to update file %q: %v", filePath, err)
+		}
+	}
+
+	pr, _, err := client.PullRequests.Create(ctx, owner, repo, &github.NewPullRequest{
+		Title: github.String(generatePRTitleFromAssets(assets)),
+		Head:  github.String(updateBranch),
+		Base:  github.String(baseBranch),
+		Body:  github.String(GeneratePRDescription(upgradePipelineRunID, buddyBuildID, assets)),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create PR: %v", err)
+	}
+
+	// This function utilizes the Issues API because in GitHub's API model, pull requests are treated as a special type of issue.
+	// While GitHub provides a dedicated PullRequests API, it doesn't currently offer a method for adding labels directly to pull requests.
+	//
+	// Therefore, we use the Issues.AddLabelsToIssue method, passing the pull request's number (which is equivalent to its issue number)
+	// to apply the labels.
+	//
+	// This approach is a workaround until GitHub potentially adds direct label management for pull requests in their API.
+	if _, _, err := client.Issues.AddLabelsToIssue(ctx, owner, repo, pr.GetNumber(), []string{"3.0-dev", "Automatic PR"}); err != nil {
+		return fmt.Errorf("error adding label to pull request: %w\n", err)
 	}
-	fmt.Println(string(txtar.Format(&ar)))
+
+	fmt.Printf("Pull request created successfully: %s\n", pr.GetHTMLURL())
 
 	return nil
 }
 
+func generateUpdateBranchNameFromAssets(assets *buildassets.BuildAssets) string {
+	return fmt.Sprintf("refs/heads/update-go-%s", assets.GoVersion().Full())
+}
+
+func generatePRTitleFromAssets(assets *buildassets.BuildAssets) string {
+	return fmt.Sprintf("Bump Go Version to %s", assets.GoVersion().Full())
+}
+
+func GeneratePRDescription(upgradePipelineRunID, buddyBuildID string, assets *buildassets.BuildAssets) string {
+	template := `Bump Go Version to %s
```

Comment:
`const template =` rather than `:=`, to allow arg mismatches to be detected by `go vet` and the linter.